### PR TITLE
fetch all computers without a date filter.

### PR DESCRIPTION
**Pull request by yedidyacohenpalo**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [CIAC-12027](https://jira-dc.paloaltonetworks.com/browse/CIAC-12027)

## Description
A few sentences describing the overall goals of the pull request's commits.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Review by jbabazadeh — CHANGES_REQUESTED** [review]
great job! please see my comments

File: Packs/JamfProtect/Integrations/JamfProtectEventCollector/JamfProtectEventCollector.py [hunk 1]
```diff
@@ -321,6 +321,87 @@ def get_computer(self, args: dict, next_page: str) -> dict:
             variables["next"] = next_page
         return self.graphql(query, variables)
 
+    def get_all_computers(self, args: dict, next_page: str) -> dict:
+        """
```
[the remark this answers is not in the document]
Comment (yedidyacohenpalo) [reply Packs/JamfProtect/Integrations/JamfProtectEventCollector/JamfProtectEventCollector.py:325]:
```suggestion
        """done
```

[the remark this answers is not in the document]
Comment (yedidyacohenpalo) [reply Packs/JamfProtect/Integrations/JamfProtectEventCollector/JamfProtectEventCollector.py:325]: done

File: Packs/JamfProtect/Integrations/JamfProtectEventCollector/JamfProtectEventCollector.yml [hunk 2]
```diff
@@ -56,6 +56,13 @@ configuration:
   type: 0
   advanced: true
   section: Collect
+- display: Fetch all computers
```
[the remark this answers is not in the document]
Comment (yedidyacohenpalo) [reply Packs/JamfProtect/Integrations/JamfProtectEventCollector/JamfProtectEventCollector.yml:59]: done

File: Packs/JamfProtect/Integrations/JamfProtectEventCollector/JamfProtectEventCollector_test.py [hunk 3]
```diff
@@ -151,22 +151,22 @@ def test_nextTrigger(
     client = Client(base_url=MOCK_BASEURL, verify=False, proxy=False, client_id=MOCK_CLIENT_ID,
                     client_password=MOCK_CLIENT_PASSWORD)
```
[the remark this answers is not in the document]
Comment (yedidyacohenpalo) [reply Packs/JamfProtect/Integrations/JamfProtectEventCollector/JamfProtectEventCollector_test.py:158]: done

File: Packs/JamfProtect/Integrations/JamfProtectEventCollector/JamfProtectEventCollector.yml [hunk 4]
```diff
@@ -56,6 +56,14 @@ configuration:
...
+  name: fetch_all_computers
+  defaultvalue: "false"
+  type: 8
+  required: false
+  advanced: true
+  section: Collect
+  additionalinfo: When set to true, retrieves all available computers during the initial fetch, ignoring the Events Fetch Interval and not limited to recent changes.
```
Comment (jbabazadeh) [inline Packs/JamfProtect/Integrations/JamfProtectEventCollector/JamfProtectEventCollector.yml:66]:
```suggestion
  additionalinfo: When set to true, retrieves all available computers during the initial fetch.
```

File: Packs/JamfProtect/Integrations/JamfProtectEventCollector/README.md [hunk 5]
```diff
@@ -16,6 +16,13 @@ Use this integration to fetch audit logs, alerts and computer events from Jamf P
     | Max alert events per fetch | Maximum number of alert events to fetch at a time. Default is 1000| False |
     | Max audit events per fetch | Maximum number of audit events to fetch at a time. Default is 20,000| False |
     | Max computer events per fetch | Maximum number of computer events to fetch at a time. Default is 1000| False |
+    | Fetch all computers | When set to true, retrieves all available computers during the initial fetch, ignoring the Events Fetch Interval and not limited to recent changes. | False |
```
Comment (jbabazadeh) [inline Packs/JamfProtect/Integrations/JamfProtectEventCollector/README.md:19]:
```suggestion
    | Fetch all computers | When set to true, retrieves all available computers during the initial fetch. | False |
```

File: Packs/JamfProtect/Integrations/JamfProtectEventCollector/README.md [hunk 6]
```diff
@@ -16,6 +16,13 @@ Use this integration to fetch audit logs, alerts and computer events from Jamf P
...
+
+
+
+
+
+
```
Comment (jbabazadeh) [inline Packs/JamfProtect/Integrations/JamfProtectEventCollector/README.md:25]:
```suggestion
```

**Review by jbabazadeh — APPROVED** [review]
looks good!

File: Packs/JamfProtect/Integrations/JamfProtectEventCollector/JamfProtectEventCollector.py [hunk 7]
```diff
@@ -472,27 +487,33 @@ def get_events_computer_type(client: Client, start_date: str, max_fetch: int, la
...
+    demisto.debug(debug_message)
+
+    client_event_type_func = client.get_computers
     events, next_page = get_events(command_args, client_event_type_func, max_fetch, next_page)
     for event in events:
         event["source_log_type"] = "computers"
+
+    latest_event = max(filter(None, (arg_to_datetime(event.get("created"), DATE_FORMAT)
+                                     for event in events))).strftime(DATE_FORMAT) if events else current_date
+
```
Comment (jbabazadeh) [inline Packs/JamfProtect/Integrations/JamfProtectEventCollector/JamfProtectEventCollector.py:503]: add debug log that print the `latest_event`

**Review by jbabazadeh — APPROVED** [review]
Nice!

**Pull request merged**